integration_test/ops_agent_test.go

```diff
@@ -2461,29 +2458,54 @@ func runResourceDetectorCli(ctx context.Context, logger *logging.DirectoryLogger
 			remote: "go.sum"},
 	}
 
+	// Create the folder structure on the VM
 	workDir := path.Join(workDirForPlatform(vm.Platform), "run_resource_detector")
+	packageDir := path.Join(workDir, "confgenerator", "resourcedetector")
+	var createFolderCmd string
+	if gce.IsWindows(vm.Platform) {
+		createFolderCmd = fmt.Sprintf("New-Item -ItemType Directory -Path %s", strings.ReplaceAll(packageDir, "/", "\\"))
+	} else {
+		createFolderCmd = fmt.Sprintf("mkdir -p %s", packageDir)
+	}
+	out, err := gce.RunScriptRemotely(ctx, logger, vm, createFolderCmd, nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create folder with %s in VM: %s", createFolderCmd, out.Stderr)
+	}
+
+	// Upload the files
 	for _, file := range filesToUpload {
 		f, err := os.Open(file.local)
 		if err != nil {
 			return nil, err
 		}
-		err = gce.UploadContent(ctx, logger, vm, f, path.Join(workDir, file.remote))
+		var remotePath string
+		if gce.IsWindows(vm.Platform) {
+			remotePath = strings.ReplaceAll(path.Join(workDir, file.remote), "/", "\\")
+		} else {
+			remotePath = path.Join(workDir, file.remote)
+		}
+		err = gce.UploadContent(ctx, logger, vm, f, remotePath)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	if err := InstallGolang(ctx, logger.ToMainLog(), vm); err != nil {
+	// Run the resource detector in the VM
+	goPath, err := InstallGolang(ctx, logger, vm)
+	if err != nil {
 		return nil, err
 	}
-
-	cmd := fmt.Sprintf("cd %s && go run run_resource_detector.go", workDir)
-	test_out, err := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", cmd)
+	cmd := fmt.Sprintf(`
+		%s
+		cd %s
+		go run run_resource_detector.go`, goPath, workDir)
```

Comment:
`workDir` hasn't had its path separators normalized yet. If that's needed then it should be added, or if it's not needed then we can probably also remove the `ReplaceAll(remotePath, "/", "\\")` from earlier to simplify.

Reply:
Removed `ReplaceAll(remotePath, "/", "\\")` as suggested.